controllers/clusterfluentbitconfig_controller.go

```diff
@@ -0,0 +1,176 @@
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+func (r *ClusterFluentBitConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = r.Log.WithValues("clusterfluentbitconfig", req.NamespacedName)
+
+	//var cfgs logging.FluentBitConfigList clusterfluentbit.ClusterFluentBitConfigList
+	var cfgs fluentbitv1alpha2.ClusterFluentBitConfigList
+	if err := r.List(ctx, &cfgs); err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	for _, cfg := range cfgs.Items {
+		// List all inputs matching the label selector.
+		var clusterInputs fluentbitv1alpha2.ClusterInputList
+		selector, err := metav1.LabelSelectorAsSelector(&cfg.Spec.ClusterInputSelector)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if err = r.List(ctx, &clusterInputs, client.MatchingLabelsSelector{Selector: selector}); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// List all filters matching the label selector.
+		var clusterFilters fluentbitv1alpha2.ClusterFilterList
+		selector, err = metav1.LabelSelectorAsSelector(&cfg.Spec.ClusterFilterSelector)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if err = r.List(ctx, &clusterFilters, client.MatchingLabelsSelector{Selector: selector}); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// List all outputs matching the label selector.
+		var clusterOutputs fluentbitv1alpha2.ClusterOutputList
+		selector, err = metav1.LabelSelectorAsSelector(&cfg.Spec.ClusterOutputSelector)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if err = r.List(ctx, &clusterOutputs, client.MatchingLabelsSelector{Selector: selector}); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// List all parsers matching the label selector.
+		var clusterParsers fluentbitv1alpha2.ClusterParserList
+		selector, err = metav1.LabelSelectorAsSelector(&cfg.Spec.ClusterParserSelector)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if err = r.List(ctx, &clusterParsers, client.MatchingLabelsSelector{Selector: selector}); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		filePath := "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+		nsRead, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		ns := string(nsRead)
+
+		// Inject config data into Secret
+		sl := plugins.NewSecretLoader(r.Client, ns, r.Log)
+		mainCfg, err := cfg.RenderMainConfig(sl, clusterInputs, clusterFilters, clusterOutputs)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		parserCfg, err := cfg.RenderParserConfig(sl, clusterParsers)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		cl := plugins.NewConfigMapLoader(r.Client, ns)
+		scripts, err := cfg.RenderLuaScript(cl, clusterFilters)
```

Comment:
The script for cluster Lua filter and namespace Lua filter are in the different directories, how to handle this?